sdk/storage/inc/blobs/blob_sas_builder.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+#include "blobs/protocol/blob_rest_client.hpp"
+#include "common/account_sas_builder.hpp"
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  enum class BlobSasResource
+  {
+    Container,
+    Blob,
+    BlobSnapshot,
+    BlobVersion,
+  };
+
+  enum class BlobContainerSasPermissions
+  {
+    None = 0,
```

Comment:
Why is it necessary to have `None` here?

Reply:
So that we can write code like this https://github.com/Azure/azure-sdk-for-cpp/blob/daa6854717c7322edfcee7d905945d0d2f2316c8/sdk/storage/src/blobs/blob_sas_builder.cpp#L14

Reply:
I can remove these `None`s if you have better way to do that.

Reply:
I usually use `if ((permissions | BlobContainerSasPermissions::Read) == permissions)`.